### PR TITLE
Feature/admin

### PR DESCRIPTION
**Pull request by FredrikMorstad**
# :sparkles: Admin page
## Page for admin updating and deleting members 
 - ## :warning: Dependant on tdctl-api pull request
 - ## Setup for adding more views such as admin and announcement 
image
## Other updates:
 - ### :fishing_pole_and_fish: Hooks
   - #### fetching events, fetching updates and update useForm
 - ### Clean up
   - #### :exclamation: fix missing dependency warning
   - #### :art: Prettier format all files

### CODE REVIEW

File: src/components/atoms/loading/loading.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
+.ring {
```
Comment (Sagensagen) [inline src/components/atoms/loading/loading.scss:1]: Nice

File: src/components/molecules/forms/registerForm/RegisterForm.tsx [hunk 2]
```diff
@@ -2,17 +2,27 @@ import { useState, useContext } from 'react';
 import useForm from 'hooks/useForm';
 import { AuthenticateContext } from 'contexts/authProvider';
 import { login, registerMember } from 'api';
-import { nameValidator, emailValidator, passwordValidator, classOfValidator, notRequiredPhoneValidator, emptyFieldsValidator } from 'utils/validators';
+import {
```
Comment (Sagensagen) [inline src/components/molecules/forms/registerForm/RegisterForm.tsx:5]: Det skjer noe funky med form'en på firefox

File: src/components/molecules/forms/settingsForm/SettingsForm.tsx [hunk 3]
```diff
@@ -56,6 +58,11 @@ const SettingsForm: React.FC<Props> = ({ init }) => {
     try {
       await updateMember(data);
       history.push('/profile');
+      addToast({
+        title: 'Suksess',
```
Comment (seblj) [inline src/components/molecules/forms/settingsForm/SettingsForm.tsx:62]: Sometimes we use the norwegian `Suksess`, and sometimes we use `Success`. We should stick to one of them and not mix

**Review by seblj — COMMENTED** [review]
Just a few minor comments since I was bored🤷‍♂️

File: src/components/molecules/forms/settingsForm/SettingsForm.tsx (hunk 3 above)
Comment (Sagensagen) [reply src/components/molecules/forms/settingsForm/SettingsForm.tsx:62]: Norwegian allerts/toasts seems to be what mostly used hence  we should use norwegian?

File: src/api/requests.ts [hunk 4]
```diff
@@ -81,6 +69,20 @@ export const put = async <T>(url: string, data: any, auth = false) => {
   return fetch(request).then((res) => handleResponse<T>(res));
 };
 
+export const Delete = async <T>(url: string, auth = false) => {
```
[the remark this answers is not in the document]
Comment (Sagensagen) [reply src/api/requests.ts:72]: `delete` is not allowed in strict mode ES5> and cannot be an identifier. Delete is an ok compromise.

[the remark this answers is not in the document]
Comment (seblj) [reply src/api/requests.ts:72]: Ah yes I forgot😅 then uppercase is ok

File: src/components/molecules/forms/settingsForm/SettingsForm.tsx (hunk 3 above)
Comment (seblj) [reply src/components/molecules/forms/settingsForm/SettingsForm.tsx:62]: I don't think I have a preference really as long as they are consistent🤷‍♂️

**Review by Sagensagen — APPROVED** [review]
Reviewed/tested most of the functionality and everything works. Some typo's fixed

**Comment by Sagensagen** [comment]
Merging

**Pull request merged**